### PR TITLE
MITAB: Fix possible crash on NULL feature text

### PR DESCRIPTION
**Pull request by drons**

### CODE REVIEW

**Comment by rouault** [comment]
would it be possible to have a test in autotest for that ?

**Comment by drons** [comment]
It takes some time to extract test case. I will try.

For now stack trace looks something like this:
```
1   __GI_raise                                                                                          raise.c              50  0x7ffff35c1ce1 
2   __GI_abort                                                                                          abort.c              79  0x7ffff35ab537 
3   ??                                                                                                                           0x7ffff37ff7ec 
4   ??                                                                                                                           0x7ffff380a966 
5   std::terminate()                                                                                                             0x7ffff380a9d1 
6   __cxa_throw                                                                                                                  0x7ffff380ac65 
7   std::__throw_logic_error(const char *)                                                                                       0x7ffff3801faa 
8   void std::string::_M_construct<const char *>(const char *, const char *, std::forward_iterator_tag)                          0x7ffff601274d 
9   CPLString::CPLString(const char *)                                                                                           0x7ffff5826d40 
10  TABText::WriteGeometryToMAPFile(TABMAPFile *, TABMAPObjHdr *, int, TABMAPCoordBlock * *)                                     0x7ffff6399f3f 
11  TABFile::WriteFeature(TABFeature *)                                                                                          0x7ffff63be002 
12  TABFile::CreateFeature(TABFeature *)                                                                                         0x7ffff63be32b 
13  IMapInfoFile::ICreateFeature(OGRFeature *)                                                                                   0x7ffff63c91dc 
14  OGRLayer::CreateFeature(OGRFeature *)                                                                                        0x7ffff69ab9cf 
15  OGR_L_CreateFeature
```

**Comment by drons** [comment]
This is minimal test case:

```c++
char	driver_name[] = "MapInfo File";
char	opt_format[] = "FORMAT=TAB";
char*	mif_opt[] = { opt_format, NULL };
std::string	fn("/tmp/x.tab");

OGRSpatialReference	osr; osr.SetFromUserInput("WGS84");
OGRPoint pt(1, 2);

OGRSFDriverH driver = OGRGetDriverByName(driver_name);
OGRDataSourceH dataset = OGR_Dr_CreateDataSource(driver, fn.data(), mif_opt);
OGRLayerH layer = OGR_DS_CreateLayer(dataset, "test", (OGRSpatialReferenceH)&osr, wkbUnknown, nullptr);
OGRFeatureH feature = OGR_F_Create(OGR_L_GetLayerDefn(layer));
OGR_F_SetStyleString(feature, "LABEL(f:DejaVu Sans,s:0.705557,c:#FF0000FF,b:#000000FF)");
OGR_F_SetGeometry(feature, &pt);
OGR_L_CreateFeature(layer, feature);

OGR_DS_Destroy(dataset);
```

The error is caused by incorrect style string `"LABEL(f:DejaVu Sans,s:0.705557,c:#FF0000FF,b:#000000FF)"`.

But it is really amazing.  I got this style string from `OGRStyleMgr::GetStyleString()` function.

I will add autotests next day.

**Comment by drons** [comment]
Test done.

**Comment by rouault** [comment]
formatting issues:
```
If you are seeing this message in CI, reproduce locally with: `pre-commit run --all-files`.
To run `pre-commit` as part of git workflow, use `pre-commit install`.
All changes made by hooks:
diff --git a/autotest/ogr/ogr_mitab.py b/autotest/ogr/ogr_mitab.py
index 92ca6aa..522df85 100755
--- a/autotest/ogr/ogr_mitab.py
+++ b/autotest/ogr/ogr_mitab.py
@@ -3063,11 +3063,13 @@ def test_ogr_mitab_read_write_hotine_oblique_mercator_with_rectified_grid_angle(
 
 def test_ogr_mitab_label_without_text():
 
-    ds = gdaltest.mapinfo_drv.CreateDataSource("/vsimem/test_ogr_mitab_label_without_text.tab")
+    ds = gdaltest.mapinfo_drv.CreateDataSource(
+        "/vsimem/test_ogr_mitab_label_without_text.tab"
+    )
     lyr = ds.CreateLayer("label")
     lyr.CreateField(ogr.FieldDefn("ID", ogr.OFTInteger))
     f = ogr.Feature(lyr.GetLayerDefn())
     f.SetGeometryDirectly(ogr.CreateGeometryFromWkt("POINT(1 2)"))
-    f.SetStyleString('LABEL(f:DejaVu Sans,s:0.705557,c:#FF0000FF,b:#000000FF)')
+    f.SetStyleString("LABEL(f:DejaVu Sans,s:0.705557,c:#FF0000FF,b:#000000FF)")
     lyr.CreateFeature(f)
     ds = None
```

File: autotest/ogr/ogr_mitab.py [hunk 1]
```diff
@@ -3055,3 +3055,21 @@ def test_ogr_mitab_read_write_hotine_oblique_mercator_with_rectified_grid_angle(
...
+    ds = gdaltest.mapinfo_drv.CreateDataSource(
+        "/vsimem/test_ogr_mitab_label_without_text.tab"
+    )
+    lyr = ds.CreateLayer("label")
+    lyr.CreateField(ogr.FieldDefn("ID", ogr.OFTInteger))
+    f = ogr.Feature(lyr.GetLayerDefn())
+    f.SetGeometryDirectly(ogr.CreateGeometryFromWkt("POINT(1 2)"))
+    f.SetStyleString("LABEL(f:DejaVu Sans,s:0.705557,c:#FF0000FF,b:#000000FF)")
+    lyr.CreateFeature(f)
+    ds = None
```
Comment (rouault) [inline autotest/ogr/ogr_mitab.py:3075]:
```suggestion
    filename = "/vsimem/test_ogr_mitab_label_without_text.tab"
    ds = gdaltest.mapinfo_drv.CreateDataSource(filename)
    lyr = ds.CreateLayer("label")
    lyr.CreateField(ogr.FieldDefn("ID", ogr.OFTInteger))
    f = ogr.Feature(lyr.GetLayerDefn())
    f.SetGeometryDirectly(ogr.CreateGeometryFromWkt("POINT(1 2)"))
    f.SetStyleString("LABEL(f:DejaVu Sans,s:0.705557,c:#FF0000FF,b:#000000FF)")
    lyr.CreateFeature(f)
    ds = None
    ogr.GetDriverByName("MapInfo File").DeleteDataSource(filename)
```

**Pull request merged**